CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/markers/PolygonMarker.java

```diff
@@ -0,0 +1,155 @@
+import javafx.scene.input.MouseEvent;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Line;
+
+/**
+ * Marker that represent the shapes the user draws
+ *
+ * @author altair1673
+ */
+public class PolygonMarker extends AbstractMarker {
+
+    private final List<Line> polygonLineUI = new ArrayList<>();
+    private Line currentLine = null;
+    private String rawPath = "";
+
+    public PolygonMarker(final MapView parent, final int markerID, final int xOffset, final int yOffset) {
+        super(parent, markerID, -99, xOffset, yOffset, AbstractMarker.MarkerType.POLYGON_MARKER);
+
+        markerPath.setStroke(Color.BLACK);
+        markerPath.setFill(Color.ORANGE);
+        markerPath.setOpacity(0.4);
+
+        // Event handler for the polygon marker
+        markerPath.setOnMouseEntered(new EventHandler<MouseEvent>() {
+            public void handle(final MouseEvent e) {
+
+                markerPath.setFill(Color.YELLOW);
+
+                e.consume();
+            }
+        });
+
+        markerPath.setOnMouseExited(new EventHandler<MouseEvent>() {
+            public void handle(final MouseEvent e) {
+
+                markerPath.setFill(Color.ORANGE);
+
+                e.consume();
+            }
+        });
+
+        markerPath.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            public void handle(final MouseEvent e) {
+
+                parent.removeUserMarker(markerID);
+                e.consume();
+            }
+        });
+    }
+
+    /**
+     * Adds a new line to the polygon marker
+     *
+     * @param prevLineEndX
+     * @param prevLineEndY
+     * @return
+     */
+    public Line addNewLine(final double prevLineEndX, final double prevLineEndY) {
+        // If current line is null it mean the first line is being drawn
+        // Else finish off the current line and add it to the UI
+        if (currentLine == null) {
+            currentLine = new Line();
+            currentLine.setStartX(prevLineEndX);
+            currentLine.setStartY(prevLineEndY);
+            currentLine.setEndX(prevLineEndX);
+            currentLine.setEndY(prevLineEndY);
+        } else {
+            currentLine.setEndX(prevLineEndX);
+            currentLine.setEndY(prevLineEndY);
+            polygonLineUI.add(currentLine);
+            currentLine = new Line();
+            currentLine.setStartX(prevLineEndX);
+            currentLine.setStartY(prevLineEndY);
+            currentLine.setEndX(prevLineEndX);
+            currentLine.setEndY(prevLineEndY);
+        }
+        currentLine.setStroke(Color.BLACK);
+        return currentLine;
+    }
+
+    /**
+     * Clear the gui that shows the polygon the user is drawing
+     */
+    public void endDrawing() {
+        currentLine = null;
+        polygonLineUI.clear();
+    }
+
+    public void setEnd(final double x, final double y) {
+        if (currentLine != null) {
+            currentLine.setEndX(x);
+            currentLine.setEndY(y);
+        }
+    }
```

Comment:
would it make sense to use this function in the function where the previous comment is?